src/boot/bootctl-status.c

```diff
@@ -130,6 +174,35 @@ static int print_efi_option(uint16_t id, int *n_printed, bool in_order) {
 
         efi_tilt_backslashes(path);
 
+        if (arg_json_format_flags != JSON_FORMAT_OFF) {
+                _cleanup_(json_variant_unrefp) JsonVariant *efi_obj = NULL;
+                char id_buffer[10];
+                snprintf(id_buffer, sizeof(id_buffer), "0x%04X", id);
+                char status_buffer[100];
+                snprintf(status_buffer, sizeof(status_buffer), "%sactive%s", active ? "" : "in", in_order ? ", boot-order" : "");
+                char partition_buffer[100];
+                snprintf(partition_buffer, sizeof(partition_buffer), "/dev/disk/by-partuuid/" SD_ID128_UUID_FORMAT_STR,
+                         SD_ID128_FORMAT_VAL(partition));
+                char file_buffer[256];
+                snprintf(file_buffer, sizeof(file_buffer), "%s", path);
+
+                r = json_variant_merge_objectb(&efi_obj, JSON_BUILD_OBJECT(
+                                    JSON_BUILD_PAIR("Title", JSON_BUILD_STRING(strna(title))),
+                                        JSON_BUILD_PAIR("ID", JSON_BUILD_STRING(id_buffer)),
```

Comment:
please use the json type system properly, i.e output integers as integers instead of formatted strings. i.e. JSON_BUILD_UNSIGNED(id)